### PR TITLE
JSONParse accepts a dot delimited string

### PR DESCRIPTION
**Pull request by se3000**

### CODE REVIEW

File: adapters/json_parse.go [hunk 1]
```diff
@@ -125,3 +127,17 @@ func isArray(js *simplejson.Json, key string) bool {
 	}
 	return true
 }
+
+type jsonPath []string
+
+func (jp *jsonPath) UnmarshalJSON(b []byte) error {
+	strs := []string{}
+	var err error
+	if utils.IsQuoted(b) {
```
Comment (j16r) [inline adapters/json_parse.go]: Under what circumstances would the incoming string not be quoted?

File: adapters/json_parse_test.go [hunk 2]
```diff
@@ -94,3 +95,31 @@ func TestJsonParse_Perform(t *testing.T) {
...
+		want      []string
+		wantError bool
+	}{
+		{"array", `{"path":["1","b"]}`, []string{"1", "b"}, false},
+		{"array with dots", `{"path":["1",".","b"]}`, []string{"1", ".", "b"}, false},
+		{"string", `{"path":"first"}`, []string{"first"}, false},
+		{"dot delimited", `{"path":"1.b"}`, []string{"1", "b"}, false},
+		{"dot delimited empty string", `{"path":"1...b"}`, []string{"1", "", "", "b"}, false},
+		{"unclosed array errors", `{"path":["1"}`, []string{}, true},
+		{"unclosed string errors", `{"path":"1.2}`, []string{}, true},
```
Comment (j16r) [inline adapters/json_parse_test.go]: Related to the question above, the implementation supports unquoted strings, but does not test them, if we feel that unquoted input should be supported I feel like we should test that.

File: utils/utils.go [hunk 3]
```diff
@@ -380,7 +380,7 @@ func StripBearer(authorizationStr string) string {
 }
 
 // IsQuoted checks if the first and last characters are either " or '.
-func isQuoted(input []byte) bool {
+func IsQuoted(input []byte) bool {
```
Comment (j16r) [inline utils/utils.go]: 👍

File: adapters/json_parse.go [hunk 4]
```diff
@@ -125,3 +127,17 @@ func isArray(js *simplejson.Json, key string) bool {
...
+		strs = strings.Split(string(utils.RemoveQuotes(b)), ".")
```
Comment (j16r) [inline adapters/json_parse.go]: Is it possible that you might ever want to match a string with a period within it?

File: adapters/json_parse.go (hunk 1 above)
Comment (se3000) [reply adapters/json_parse.go]: If it is an array, the first character won't be quotes.

File: adapters/json_parse.go (hunk 4 above)
Comment (se3000) [reply adapters/json_parse.go]: If you want that, you can always use the already existing array syntax; e.g. `{"path": ["a.", "b."]}`

File: adapters/json_parse_test.go (hunk 2 above)
Comment (se3000) [reply adapters/json_parse_test.go]: I don't think it will support unquoted strings, but can write a test for that if needed.

Comment (se3000) [reply adapters/json_parse_test.go]: Yeah, unquoted strings error out.

Comment (j16r) [reply adapters/json_parse_test.go]: I don't believe JSON supports unquoted strings, so what would it be processing?

Comment (se3000) [reply adapters/json_parse_test.go]: Idk, maybe I misinterpreted what you thought it'd be doing. Sounds like we agree, no need to support unquoted input.

**Pull request merged**